Modules/Quotas/src/Quotas.Application/IntegrationEvents/Incoming/IdentityCreated/IdentityCreatedIntegrationEventHandler.cs

```diff
@@ -4,22 +4,34 @@
 using Microsoft.Extensions.Logging;
 
 namespace Backbone.Modules.Quotas.Application.IntegrationEvents.Incoming.IdentityCreated;
+
 public class IdentityCreatedIntegrationEventHandler : IIntegrationEventHandler<IdentityCreatedIntegrationEvent>
 {
     private readonly IIdentitiesRepository _identitiesRepository;
+    private readonly ITiersRepository _tiersRepository;
     private readonly ILogger<IdentityCreatedIntegrationEventHandler> _logger;
 
-    public IdentityCreatedIntegrationEventHandler(IIdentitiesRepository identitiesRepository, ILogger<IdentityCreatedIntegrationEventHandler> logger)
+    public IdentityCreatedIntegrationEventHandler(IIdentitiesRepository identitiesRepository, ILogger<IdentityCreatedIntegrationEventHandler> logger, ITiersRepository tiersRepository)
     {
         _identitiesRepository = identitiesRepository;
         _logger = logger;
+        _tiersRepository = tiersRepository;
     }
 
     public async Task Handle(IdentityCreatedIntegrationEvent integrationEvent)
     {
+        _logger.LogTrace("Handling IdentityCreatedIntegrationEvent ...");
+
         var identity = new Identity(integrationEvent.Address, integrationEvent.Tier);
         await _identitiesRepository.Add(identity, CancellationToken.None);
 
         _logger.LogTrace($"Successfully created identity. Identity Address: {identity.Address}, Tier ID: {identity.TierId}");
+
+        var tier = await _tiersRepository.Find(identity.TierId, CancellationToken.None);
+
+        tier.Quotas.ForEach(identity.AssignTierQuotaFromDefinition);
+        await _identitiesRepository.Update(identity, CancellationToken.None);
```

Comment:
Hmm, why don't you just add the tiers BEFORE you call `_identitiesRepository.Add`? Because currently you need to database write operations.

Reply:
That is a very good point, I failed to look at the current handler logic and see if it could be adjusted, and simply focused on adding the additional logic.